### ISSUE TITLE
[BUG] Binary-compatible upgrade documentation

### ISSUE THREAD

**Issue by djbpitt** [eXist-db/documentation#413]
**Describe the bug**
Documentation of binary-compatible upgrade at <http://exist-db.org/exist/apps/doc/upgrading> could be improved. Tested by upgrading from 5.0 to 5.1.1 on MacOS using dmg installer.

The documentation advises removing the empty new *data* directory and copying the old one in its place. It also says that the default location of the *data* directory is*$EXIST_HOME/data*. On my system *$EXIST_HOME* is not set and eXist-db installed itself into */Applications*, while the data installed itself into */Users/my_user_id/Library/Application Support/org.exist* (without the directory name *data*, that is, there is no directory called *data*). The new installation defaulted to use that same location for data, so there was no empty directory to delete and nothing to copy.

**Comment by duncdrum** [comment]
Just out of curiosity what is your macOS version?

**Comment by djbpitt** [comment]
@duncdrum Mojave

**Comment by dizzzz** [comment]
@djbpitt you are correct; the description does match the a-typical directory layout on macOS.

`$EXIST_HOME` is probably confusing as I think $EXIST_HOME is now more a 'symbolic representation` of a directory that contains on non-Mac systems both data and binaries. it is not the environment variable (anymore), I for now, I would have no idea how to call it different.

And the $EXIST_HOME variable might still work,  not sure.

**Comment by duncdrum** [comment]
@joewiz could you confirm that there is no `exist-data` directory with `.dmg` installs? this sounds like a proper bug to me. 

@djbpitt i think there are two issues here, one is the missing `exist-data` everything in `org.exist/` sounds disconcerting. The other is the use of `$EXIST_HOME` throughout the docs, which no longer has the same meaning or utility as it had up until v4.

**Comment by joewiz** [comment]
@duncdrum The eXist-db.app supplied by the DMG distribution contains an empty `data` directory: `Contents/Resources/data`. However, upon starting up eXist-db.app, the launcher displays a dialog asking you to select a data directory; by default, it suggests `/Users/{current-user}/Library/Application Support/org.exist`.

**Screenshot of `eXist-db.app/Contents/Resources/data`**

> ![Screen Shot 2020-01-28 at 5 12 32 PM](https://user-images.githubusercontent.com/59118/73310218-da15b180-41f1-11ea-919f-393d4805f176.png)

**Screenshot of launcher configuration window** (showing default data directory location)

> ![Screen Shot 2020-01-28 at 5 14 16 PM](https://user-images.githubusercontent.com/59118/73310279-fade0700-41f1-11ea-8628-3e4f232f2fec.png)

**Screenshot of `/Users/{current-user}/Library/Application Support/org.exist`**

> ![Screen Shot 2020-01-28 at 5 15 28 PM](https://user-images.githubusercontent.com/59118/73310256-ec8feb00-41f1-11ea-9cff-031dae738bc2.png)
